src/common.fypp

```diff
@@ -113,4 +113,33 @@ ${prefix + joinstr.join([line.strip() for line in txt.split("\n")]) + suffix}$
   $:"{0}_{1}_{2}{3}_{2}{3}".format(gname, rank, type[0], kind) if suffix == '' else "{0}_{1}_{2}{3}_{4}".format(gname, rank, type[0], kind, suffix)
 #:enddef
 
+#! Generates Fortran expressions.
+#!
+#! Args:
+#!   varname (str): Name of the variable to be used as origin
+#!   varname1 (str): Name of the variable to be used instead of varname
+#!   origrank (int): Rank of the original variable
+#!   dim (int): Index of the used expression varname1
+#!
+#! Returns:
+#!   Shape expression enclosed in braces, except for the index dim.
+#!
+#! E.g., (:, :, :, i, :, :)
+#!
+
+#:def rankindice(varname, varname1, origrank, dim)
```

Comment:
We should probably rename this macro to have a more descriptive name.  If it is only used to select subarrays by reducing the dimension, we could have:
```
#:def select_subarray(origrank, selectors)
  #:assert origrank > 0
  #:set seldict = dict(selectors)
  #:call join_lines(joinstr=", ", prefix="(", suffix=")")
    #:for i in range(1, origrank + 1)
      $:seldict.get(i, ":")
    #:endfor
  #:endcall
#:enddef
```
and use it as
```
#!  -> x(:, i, :)
x${select_subarray(3, [(2, 'i')])}$
```
It could also be used, if we need to reduce more than one rank, e.g.
```
#!  -> x(:, :, i, j)
x${select_subarray(4, [(3, 'i'), (4, 'j')])}$
```

Also the description should be clarified a bit.

Reply:
Implemented as suggested. The proposed macro is more general and better fit to its aim.
Could you have another review, please?